### ISSUE TITLE
view dom when take photo

### ISSUE THREAD

**Issue by jedofus** [cordova-plugin-camera-preview/cordova-plugin-camera-preview#270]
Hi,
i want to have my DOM HTML when i take photo.
But when i take photo, my DOM is hide. How can i resolved it ? 
I want to upload photo on my server, how can i convert base64 to URI image ? 

Thank you !

**Comment by anneb** [comment]
This plugin should enable you to combine DOM with both camera preview and picture taken. See the example application and read the documentation about showing the photo taken.

About the server side conversion: that really depends on the server side software. If you happen to have nodejs on your server, you could use:

On the browser / client:
```
function uploadPhotoData(base64Imagedata, callback) {
      var xhr = new XMLHttpRequest();
      var formData = 'photo=' + encodeURIComponent(base64Imagedata); 
      xhr.open('POST', 'http://yourserver/uploadphoto');
      xhr.setRequestHeader('Content-Type','application/x-www-form-urlencoded; charset=UTF-8');
      xhr.onreadystatechange = function (event) {
         if (xhr.readyState === 4) {
             if (xhr.status === 200) {
               callback (null, xhr.responseText);
             } else {
                callback('Error', xhr.statusText);
             }
         }
      };
      xhr.send(formData);
  };

```

On the nodejs server:
```
var express = require('express');
var fs = require('fs-extra');
var bodyParser = require('body-parser');
var cors = require('cors');
var app = express();

app.use(bodyParser.urlencoded({extended: true, limit: '50mb'}));
app.post('/uploadphoto', cors(), function(req, res) {
    fs.writeFile('/tmp/newphoto.jpg', new Buffer(req.body.photo, 'base64'), 'binary', function(err) {
         if (err) {
               res.end('Server error: ', err);
         } else {
               res.end('Photo received');
         }
   }
)
```

**Comment by jedofus** [comment]
hello, thank you for your response .
I could not find the code to take pictures with the dom. 
I can see my DOM when previewing but when i take photo, my DOM is hide and i have no solution to combine this.

**Comment by mgrubinger** [comment]
Hi @jedofus,

1. make sure you have ```toBack: true``` in the options for ```startCamera```
2. This will allow you to have the camera "in the background", therefore you have to make sure your <body> has ```background-color: transparent```. Also make sure you don't have any other elements that might cover the camera image.

About taking photos: which part of https://github.com/cordova-plugin-camera-preview/cordova-plugin-camera-preview#takepictureoptions-successcallback-errorcallback causes trouble?

**Comment by jedofus** [comment]
Hi thank you @mgrubinger. I resolved my problem.
Now i have a problem with landscape orientation. camera preview not 100% width screen. 
How can i resolved it ? #275
